### ISSUE TITLE
Create ALB instead of NLB - session stickiness etc.

### ISSUE THREAD

**Issue by perry-mitchell** [kubernetes-sigs/aws-load-balancer-controller#3079]
Hi, I've been searching for some time now and haven't found a way to create an Application Load Balancer rather than a Network Load Balancer. Is there somewhere in your docs that states whether or not this is possible or what the reasoning around only supporting NLBs?

I'd prefer to use the ALB's sticky sessions configuration rather than trying to figure out the NLB's more limited option. I've also been unable to reliably configure the NLB's sticky sessions with the following annotations (I'm using terraform):

Ingress:
```
{
    "alb.ingress.kubernetes.io/target-type" : "ip",
    "alb.ingress.kubernetes.io/target-group-attributes" : "stickiness.enabled=true,stickiness.type=app_cookie,stickiness.app_cookie.cookie_name=ld-lc-session,stickiness.app_cookie.duration_seconds=86400"
    "alb.ingress.kubernetes.io/certificate-arn" : aws_acm_certificate.app.arn,
    "alb.ingress.kubernetes.io/ssl-policy" : "ELBSecurityPolicy-FS-1-2-Res-2020-10",
    "alb.ingress.kubernetes.io/listen-ports" : jsonencode([
      {
        "HTTP" = 80
      },
      {
        "HTTPS" = 443
      }
    ])
}
```

Service:
```
{
    "service.beta.kubernetes.io/aws-load-balancer-type": "nlb-ip",
    "service.beta.kubernetes.io/aws-load-balancer-ssl-cert" : aws_acm_certificate.app.arn,
    "service.beta.kubernetes.io/aws-load-balancer-ssl-ports" : "443"
}
```

Ultimately my issue boils down to 2 points:
1. Can I use an ALB instead of an NLB?
2. If not, what is the correct configuration for sticky sessions on an NLB?

**Comment by perry-mitchell** [comment]
Ok, I managed to get ALBs to be created by simply setting the annotation `"kubernetes.io/ingress.class": "alb"` on the ingress. Not sure if this is mentioned in the docs somewhere?

Problem now is that the naming is wrong. If I set `"service.beta.kubernetes.io/aws-load-balancer-name": "livechat-staging"` I get `k8s-livechatstagingin-2dfc5ac308` in AWS being created..

**Comment by perry-mitchell** [comment]
Ok, managed to get everything solved. I would recommend revising the docs as they've not been clear with regards to ALB configuration (they're angled towards NLBs). ALBs are superior in terms of session stickiness imo.

Basically I had to pass the subnets to the ingress too, not just the service - this is also confusing, as there are several annotations that can be passed to both, and its not clear when it's required.

Listeners were created only when I used `path = "/*"` in the spec of my ingress.
